### PR TITLE
Typescript support fixes.

### PR DESCRIPTION
**Pull request by melphis**

### CODE REVIEW

File: lib/binary_parser.ts [hunk 1]
```diff
@@ -6,22 +6,22 @@ const aliasRegistry: { [key: string]: Parser } = {};
...
-  encoding?: 'utf8';
-  lengthInBytes?: null;
+  length?: number | string | ((item: any) => number);
+  assert?: number | string | ((item: any) => any);
+  lengthInBytes?: number | string | ((item: any) => number);
+  type?: string | Parser;
+  formatter?: (item: any) => string | number;
+  encoding?: string;
   readUntil?: 'eof';
   greedy?: null;
```
Comment (keichi) [inline lib/binary_parser.ts]: greedy should be a boolean

File: lib/binary_parser.ts [hunk 2]
```diff
@@ -6,22 +6,22 @@ const aliasRegistry: { [key: string]: Parser } = {};
...
 
 interface ParserOptions {
-  length?: number;
-  type?: Types | Parser;
-  assert?: (item: any) => void | string | number;
-  formatter?: (item: any) => void;
-  encoding?: 'utf8';
-  lengthInBytes?: null;
+  length?: number | string | ((item: any) => number);
+  assert?: number | string | ((item: any) => any);
```
Comment (keichi) [inline lib/binary_parser.ts]: assert accepts a function that returns a boolean

File: lib/binary_parser.ts [hunk 3]
```diff
@@ -6,22 +6,22 @@ const aliasRegistry: { [key: string]: Parser } = {};
...
+  choices?: { [key: number]: string | Parser };
+  defaultChoice?: string | Parser;
   zeroTerminated?: boolean;
   clone?: null;
   stripNull?: null;
   key?: null;
-  tag?: null;
-  offset?: null;
+  tag?: string;
+  offset?: number;
```
Comment (keichi) [inline lib/binary_parser.ts]: offset expects `number | string | ((item: any) => number)`
See https://github.com/keichi/binary-parser#pointername-options

**Comment by melphis** [comment]
Merged with last master commit.

**Pull request merged**